infrastructure/merklemountainrange/src/merkle_change_tracker.rs

```diff
@@ -190,16 +282,45 @@ impl MerkleChangeTracker {
         if !self.enabled {
             return Ok(());
         }
+        hashmap.drain();
```

Comment:
I don't think this does anything. You still need to call `collect()` or `take()` on this struct